src/EditorFeatures/Core/Workspaces/ITextBufferVisibilityTracker.cs

```diff
@@ -37,8 +38,11 @@ internal interface ITextBufferVisibilityTracker
     internal static class ITextBufferVisibilityTrackerExtensions
     {
         /// <summary>
-        /// Waits the specified amount of time while the specified <paramref name="subjectBuffer"/> is not visible.  If any
-        /// document visibility changes happen, the delay will cancel.
+        /// Waits the specified amount of time while the specified <paramref name="subjectBuffer"/> is not visible.  If
+        /// any document visibility changes happen, the delay will cancel. Note: in the event of cancellation, the
+        /// returned task will transition to the <see cref="TaskStatus.RanToCompletion"/> state not the <see
+        /// cref="TaskStatus.Canceled"/> state. This is to prevent excess exception throws in the common case of
+        /// cancellation.  Callers should check if they are canceled after this returns and bail out gracefully.
         /// </summary>
         public static async Task DelayWhileNonVisibleAsync(
```

Comment:
📝 The general approach here is to retain exactly the original behavior for the caller, but instead of _throwing_ the cancellation exception, we transition the returned `Task` instance to the canceled state without actually throwing the exception. I would prefer to continue with that general approach here instead of introducing a new pattern that callers will not expect to need to deal with.

Reply:
changed to support this approach.  lmk if there's a cleaner way to do it.